Tests/Functional/AbstractQueueTest.php

```diff
@@ -179,16 +179,32 @@ public function abortRemovesMessageFromActiveQueue()
         $this->queue->waitAndReserve(1);
 
         $this->queue->abort($messageId);
-        $this->assertSame(0, $this->queue->count());
+        $this->assertSame(0, $this->queue->countReady());
         $this->assertNull($this->queue->waitAndTake(1));
     }
 
     /**
      * @test
      */
-    public function countReturnsZeroByDefault()
+    public function countReadyReturnsZeroByDefault()
     {
-        $this->assertSame(0, $this->queue->count());
+        $this->assertSame(0, $this->queue->countReady());
+    }
+
+    /**
+     * @test
+     */
+    public function countFailedReturnsZeroByDefault()
+    {
+        $this->assertSame(0, $this->queue->countFailed());
+    }
```

Comment:
We should add tests effectively covering the new features:
`countFailedReturnsNumberOfFailedMessages()` and `countReservedReturnsNumberOfReservedMessages()`

Reply:
Added with bdffb4a5cef61544aeaffd999d85a9c6db799190